todo-app/src/stores/TodosStore.js

```diff
@@ -0,0 +1,70 @@
+import todos from "../data/todos.json";
+import { observable, action, configure } from "mobx";
+
+configure({ enforceActions: true });
+
+
+const TodoItems = observable({
+
+    todos,
+
+    get nextId() {
+        if (!this.todos.length) {
+            return 1;
+        }
+        const todoIDs = this.todos.map(todo => todo.id);
+        return Math.max(...todoIDs) + 1
+    },
+
+    addTodo: action((id, description, deadline) => {
+        if (description && deadline) {
+            const newTodo = {
+                id,
+                description,
+                deadline,
+                done: false, 
+                editMode: false
+            }
+            TodoItems.todos.push(newTodo);
+        }
+    }),
+
+    toggleCheck: action(todoID => {
```

Comment:
Looks to me like your `toggleCheck` and `toggleEdit` could be abstracted into a single function, eg `toggleProperty`.